zebra-chain/src/primitives/zcash_history/tests/vectors.rs

```diff
@@ -11,17 +11,17 @@ use eyre::Result;
 /// and its next block.
 #[test]
 fn tree() -> Result<()> {
-    tree_for_network_upgrade(Network::Mainnet, NetworkUpgrade::Heartwood)?;
-    tree_for_network_upgrade(Network::Testnet, NetworkUpgrade::Heartwood)?;
-    tree_for_network_upgrade(Network::Mainnet, NetworkUpgrade::Canopy)?;
-    tree_for_network_upgrade(Network::Testnet, NetworkUpgrade::Canopy)?;
+    tree_for_network_upgrade(&Network::Mainnet, NetworkUpgrade::Heartwood)?;
+    tree_for_network_upgrade(&Network::Testnet, NetworkUpgrade::Heartwood)?;
+    tree_for_network_upgrade(&Network::Mainnet, NetworkUpgrade::Canopy)?;
+    tree_for_network_upgrade(&Network::Testnet, NetworkUpgrade::Canopy)?;
     Ok(())
 }
 
-fn tree_for_network_upgrade(network: Network, network_upgrade: NetworkUpgrade) -> Result<()> {
+fn tree_for_network_upgrade(network: &Network, network_upgrade: NetworkUpgrade) -> Result<()> {
     let (blocks, sapling_roots) = network.block_sapling_roots_map();
 
-    let height = network_upgrade.activation_height(network).unwrap().0;
+    let height = network_upgrade.activation_height(&network).unwrap().0;
```

Comment:
Looks like a double borrow.
```suggestion
    let height = network_upgrade.activation_height(network).unwrap().0;
```

Reply:
change applied